solidstart-tanstackquery-tailwind-modules/src/services/get-user-profile.ts

```diff
@@ -2,15 +2,15 @@ import FetchApi, { ApiProps } from './api';
 import { useAuth } from '../auth';
 import { GITHUB_GRAPHQL } from '../utils/constants';
 import { USER_PROFILE_QUERY } from './queries/user-profile';
-import { UserProfile } from '~/types/user-profile-type';
+import { UserProfileProps } from '~/types/user-profile-type';
 
 type UserProfileVariables = {
   username: string;
 };
 
 type Response = {
   data: {
-    user: UserProfile;
+    user: UserProfileProps;
```

Comment:
So in general, the props of a component, or params of a function call, would (or can) be a different type than the entity type itself.

The `user` should probably be either type `User` or maybe `Profile` or `UserProfile` but it being `UserProfileProps` doesn't seem right.

If you had a `UserProfile` type, and the `Profile` components PropsType happened to exactly match the `UserProfile` type, than I would possible just have the Props be type as a `UserProfile` but the other way around seems backwards

Reply:
Sure will update it.